packages/core/core-flows/src/definition/cart/steps/reserve-inventory.ts

```diff
@@ -0,0 +1,51 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IInventoryServiceNext } from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  items: {
+    id?: string
+    inventory_item_id: string
+    required_quantity: number
+    allow_backorder: boolean
+    quantity: number
+    location_ids: string[]
+  }[]
+}
+
+export const reserveInventoryStepId = "reserve-inventory-step"
+export const reserveInventoryStep = createStep(
+  reserveInventoryStepId,
+  async (data: StepInput, { container }) => {
+    const inventoryService = container.resolve<IInventoryServiceNext>(
+      ModuleRegistrationName.INVENTORY
+    )
+
+    const items = data.items.map((item) => ({
+      line_item_id: item.id,
+      inventory_item_id: item.inventory_item_id,
+      quantity: item.required_quantity * item.quantity,
+      allow_backorder: item.allow_backorder,
+      location_id: item.location_ids[0],
```

Comment:
Why does the step accept multiple location IDs, if we always use only one?

Reply:
Currently this is the behavior, [to get first location](https://github.com/medusajs/medusa/blob/884428a1b573e499d7659aefed639bf797147428/packages/medusa/src/services/product-variant-inventory.ts#L539).
But I believe this should be reviewed to allocate the stock in all available locations until it matches the quantity.

Reply:
Yeah I agree. We might even need to look into stock allocation strategies, so merchants can configure exactly how an order is fulfilled and from where